mongomock/aggregate.py

```diff
@@ -984,6 +984,28 @@ def _find_matches_for_depth(query):
                 found_items.add(new_match['_id'])
         return new_matches
 
+    def _recursive_get(match, nested_fields):
+        first_field = nested_fields[0]
+        head = match.get(first_field)
+        if len(nested_fields) == 1:
+            # final/last field reached
+            yield head
+        else:
+            # more to go, get remaining from list/dict
+            remaining_fields = nested_fields[1:]
```

Comment:
For better readability I suggest that you move this before line 991 and check remaining_fields instead:

```py
head = match.get(nested_fields[0])
remaining_fields = nested_fields[1:]
if not remaining_fields:
  yield head
```